salt/tests/unit/modules/test_metalk8s_kubernetes.py

```diff
@@ -0,0 +1,566 @@
+                    )
+                else:
+                    raise ApiException(
+                        status=raise_retrieve,
+                        reason='An error has occurred'
+                    )
+
+            res = MagicMock()
+            # Do not return a real object as it does not bring any value in
+            # this test
+            res.to_dict.return_value = "<{} object dict>".format(name)
+            return res
+
+        get_kind_info_mock = MagicMock()
+        if not info_scope:
+            get_kind_info_mock.side_effect = ValueError(
+                'An error has occurred'
+            )
+        get_kind_info_mock.return_value.scope = info_scope
+
+        retrieve_mock = get_kind_info_mock.return_value.client.retrieve
+        retrieve_mock.side_effect = _retrieve_mock
+
+        manifest_read_mock = MagicMock()
+        # None = IOError
+        # False = YAMLError
+        if manifest_file_content is None:
+            manifest_read_mock.side_effect = IOError(
+                'An error has occurred'
+            )
+        elif manifest_file_content is False:
+            manifest_read_mock.side_effect = yaml.YAMLError(
+                'An error has occurred'
+            )
+        else:
+            manifest_read_mock.return_value = manifest_file_content
+
+        utils_dict = {
+            'metalk8s_kubernetes.get_kind_info': get_kind_info_mock
+        }
+        salt_dict = {
+            'metalk8s_kubernetes.read_and_render_yaml_file': manifest_read_mock
+        }
+        with patch.dict(metalk8s_kubernetes.__utils__, utils_dict), \
+                patch.dict(metalk8s_kubernetes.__salt__, salt_dict):
+            if raises:
+                self.assertRaisesRegexp(
+                    Exception,
+                    result,
+                    metalk8s_kubernetes.get_object,
+                    **kwargs
+                )
+            else:
+                self.assertEqual(
+                    metalk8s_kubernetes.get_object(**kwargs),
+                    result
+                )
+                retrieve_mock.assert_called_once()
+                if called_with:
+                    self.assertDictContainsSubset(
+                        called_with,
+                        retrieve_mock.call_args.kwargs
+                    )
+
+    @parameterized.expand(
+        param.explicit(kwargs=test_case)
+        for test_case in YAML_TESTS_CASES['update_object'] + YAML_TESTS_CASES['common_tests']
+    )
+    def test_update_object(self, result, raises=False, initial_obj=None,
+                           info_scope="cluster", manifest_file_content=None,
+                           called_with=None, **kwargs):
+        """
+        Tests the return of `update_object` function
+        """
+        def _update_mock(body, **_):
+            if not initial_obj:
+                raise ApiException(status=0, reason='An error has occurred')
+
+            print(body)
+            print(initial_obj)
```

Comment:
I assume that it must be a debug leftover, no ?

Reply:
No, not at all :roll_eyes: :smile: